### ISSUE TITLE
Making audio streaming a first class citizen - Ideas for an "Audio Mode" UI

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#2610]
Plenty of people use Newpipe as an audio streaming app. Right now, all the features are oriented towards video watching and there is an option to play audio on a somewhat individual basis (one video, one playlist, one channel). 

What I propose is an "Audio Mode", so that all the UI actions prefer to affect and interact with audio streams.

There would be either a headphone icon in the main toolbar, or the option "Audio Mode" in the 3 dot menu/side menu (I prefer 3 dot because it is available in all views). It would cause/require the following changes:
- Separate Subscriptions list that the user can put audio oriented channels in (music, podcasts, etc.). So two Subscriptions databases, one shown in Video Mode like it is currently, one in Audio Mode. The What's New tab would change accordingly, populated with the latest audios from the subscribed channels in Audio Mode.
- Separate Bookmarked Playlists for Audio Mode.
- Maybe separate History as well?
- Changes to the video (now audio) details page:
  - the resolution selector would change to a bitrate selector instead. (This requires adding all available audio bitrates as discussed in #1059)
  - the Background Playback headphone icon would be replaced by a Foreground Playback video (TV?) icon for loading the video stream as well for that particular URI.
  - the Popup icon could be replaced by Add to Queue, quite a common action for the audio streaming use case.
- Changes to the Background Player UI: It has two elements - the queue, and the actual player UI (title, controls, seekbar). I wish to take these elements and bring them to the foreground of all the screens during playback:
  - Playing an audio would cause the actual player UI to show up at the bottom.
  - Swiping up on it would cause it to float to the top, with the queue showing up below it.
  - This player UI should stay visible on most pages/views, with obvious exceptions like the app settings menu. This would allow the user to freely browse audio details, lists, comments, and so on.
  - Tapping on the playing audio's title would take the user to the audio details page.
- Wherever there is a list of audios (search results, What's New, Trending, etc.), the thumbnail could contain a translucent Play icon.
  - Tapping on the thumbnail would cause that audio to play immediately.
  - Tapping elsewhere on the entry would open the audio details page like normal.

Audio Mode could especially benefit from the UI changes being proposed in #2583 as well.

Ideally, Audio Mode would be the default and _only_ mode for audio only services like Soundcloud.

(... yes, I had entirely too much free time today.)

**Comment by floss-birdy** [comment]
Cool idea , to add to this i  can suggest a audio quality choice while streaming

**Comment by opusforlife2** [comment]
@floss-birdy That's what bitrate selector means above.

**Comment by OldGuyInTheClub** [comment]
I found this idea while about to suggest something similar.  I love NewPipe and queue up a lot of items even though I am listening to the audio while walking, driving, etc.  As I understand it, the video comes down anyway which adds to data usage.  Having a way to select audio only within the current NewPipe structure would be terrific.

**Comment by Stypox** [comment]
> the video comes down anyway which adds to data usage

That's not true, NewPipe only downloads the audio if you add videos to a background queue

**Comment by opusforlife2** [comment]
Whoaaaaa. This talk about queueing just made me realise - is there a way to add a single video to a _Video Queue?_

There is an Audio Queue, and a Pop-up Queue. There is also a Playlist Queue. However, to my knowledge, there is no Video Queue.

If you can confirm there is none, I'll open an issue for it.

**Comment by wb9688** [comment]
@opusforlife2: There is a video queue:

![Screenshot_20200517-150628_NewPipe](https://user-images.githubusercontent.com/46277131/82147751-255ed900-9850-11ea-8754-fc0cc45fe31e.jpg)

AFAIK the only way to add videos there is to switch to pop-up/background, add them to that queue, and then switch back to main. Well, you could of course also tap on play all in playlists.

**Comment by OldGuyInTheClub** [comment]
> > the video comes down anyway which adds to data usage
> 
> That's not true, NewPipe only downloads the audio if you add videos to a background queue

Thanks for the clarification.  I am happy to learn this.

**Comment by opusforlife2** [comment]
@wb9688 That is what I meant by Playlist Queue. It isn't manually editable without the workaround you gave. I'll open an issue.

**Comment by xplosionmind** [comment]
Hi! Are there **any news** about using NewPipe as more audio streaming focused? I'd love to use it, just to listen to music on YouTube, but without having to keep the screen on etc, as mentioned in the post at the beginning.

I really need something like this since I have no money to pay for Spotify. I tried using [MusicPiped](https://github.com/deep-gaurav/MusicPiped) and [Dingo](https://github.com/AbelTesfaye/dingo), but they're full of bugs and not much in active development.

@opusforlife2, if you still use NewPipe mainly music streaming app, which settings do you suggest me to set? **What’s your current workflow for listening to music from YouTube and NewPiped**?

Thanks a lot!

Tommi

**Comment by opusforlife2** [comment]
Currently, you can use an audio streaming service like Soundcloud, or use the search filter menu in Youtube to search specifically on Youtube Music. There is no specific setting in Newpipe for audio only use cases.

Newpipe will eventually get support for Youtube mixes and (hopefully) Youtube Music mixes. Those will help later.

**Comment by MD77MD** [comment]
>Whoaaaaa. This talk about queueing just made me realise - is there a way to add a single video to a Video Queue?

@opusforlife2  really... i have complained about this so many times... i wish you guys would give a little bit more attention to my comments 😔

**Comment by opusforlife2** [comment]
I've already pointed you to the issue I opened about it, earlier. Pay more attention to _my_ comments. ;-)

**Comment by xplosionmind** [comment]
> or use the search filter menu in Youtube to search specifically on Youtube Music.

I hope this isn't off-topic: I'm a NewPipe noob and I'm still getting used to taking full advantage of it without losing time every time to use a specific feature. Could you suggest a useful video tutorial/documentation page which thoroughly explains how to do this?







> There is no specific setting in Newpipe for audio only use cases.

I believe this is what this issue is aiming to, right?

**Comment by opusforlife2** [comment]
> I believe this is what this issue is aiming to, right?

Yup.

You don't need a tutorial. When you're typing in the search bar, tap on the 3 dot menu and voila, you have the option to search for songs, etc. on Youtube Music.

**Comment by MD77MD** [comment]
>I've already pointed you to the issue I opened about it, earlier. Pay more attention to my comments. ;-)

@opusforlife2 actually i was trying for more compassion because i was a bit down by your comment but i think you don't get it... anyway what I was trying to say that I've talked about this before but you never said anything and now all of a sudden you're acting surprised... it would have been nice if you opened the issue then.

and believe it or not I do read your comments carefully... infact everyone's comments before I say something... but don't say anything from fear of been rejected so many times...🤕

P.S: i really like and encourage the audio streaming makeover as many of newPipe users use this feature extensively... I think this should be give priority in newPipe future releases.

**Comment by opusforlife2** [comment]
Why were you a bit down by my comment? Look at the date when I opened that issue. It was months ago, in May. You talked about opening a similar issue 10 days ago: https://github.com/TeamNewPipe/NewPipe/pull/3178#issuecomment-693691892. When I saw your comment about queuing videos on the main player _one_ day after that (https://github.com/TeamNewPipe/NewPipe/pull/3178#issuecomment-694708569), I linked you to my issue, which, again, I had opened months ago. What is this talk about ignoring, dude?

**Comment by MD77MD** [comment]
@opusforlife2 
I don't know what to say other then I'm sorry...

i never saw you linking me because stupid github doesn't show the whole thread (because I'm using the mobile version) and i don't use the bill notification all the time... i guess I'm going to have to use the bill only from now on...

btw... how the hell did you find my comment https://github.com/TeamNewPipe/NewPipe/pull/3178#issuecomment-694708569 ... i have tried so much to do so but couldn't... please tell  the trick on how to filter all my comments. 
I used this[comments: MD77MD] but didn't give what i really want.

**Comment by opusforlife2** [comment]
I found it by checking the date on which I had mentioned 3619 in 3178, which led me to my own comment. Yours was just a few comments above. No issues, man. Things like this happen.

**Comment by JPGuillemin** [comment]
When can we expect this to be implemented??

**Comment by opusforlife2** [comment]
It's a very huge change, and will also likely only be used by a subset of users. So the likelihood is quite low, unless some developer who is enthusiastic about this contributes a PR.

**Comment by JPGuillemin** [comment]
Making the "play in background" player option permanent as a global preference is a very small and easy change. 

I'm not android dev, but I'm a developer: even with more complex languages like C it would be easy to replace a widget button state property by a global variable set from preferences.

**Comment by opusforlife2** [comment]
Then you can open a new issue for this as I suggested earlier.

**Comment by RSoulwin** [comment]
i really hope someone checks this out for implementation "Making audio streaming a first class citizen"
